### ISSUE TITLE
Variables from collection are not replaced in "Generate Code"

### ISSUE THREAD

**Issue by Salokyn** [usebruno/bruno#3127]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

1. I have an environnent with variables
2. I have a collection with headers using the variables from the environment and basic auth using user/password from environment

![image](https://github.com/user-attachments/assets/b34cdbd2-9670-4479-a42f-03cdc127677f)
![image](https://github.com/user-attachments/assets/700fa7eb-65c3-4c64-a0ca-0a18c702ec6a)

4. In the bru files, no headers are set and auth is set to "Inherit"

![image](https://github.com/user-attachments/assets/d5ac8ae4-f024-4252-84fb-f1d0a76c5489)

5. When I "Generate Code", variable in headers is not replaced by its value and "Authorization" header is wrong (it's Base64 of string "{{user}}:{{password}}")

```
curl --request GET \
  --url 'http://exemple.com/api/platform/444/beneficiaire/?query=2' \
  --header 'Authorization: Basic e3t1c2VyfX06e3twYXNzd29yZH19' \
  --header 'X-CD13-Email: {{x-cd13-email}}' \
  --header 'X-CD13-IP: 123.123.123.123'
```

### Screenshots/Live demo link

See above

**Comment by davidsarosap** [comment]
I can confirm that.
{{serNameGen}} is substituted by the code itself and the JS template literal isn't evaluated.
![image](https://github.com/user-attachments/assets/bcb45d72-0146-404a-a3f4-4d79d6f927b0)

**Comment by mdm** [comment]
Looking at the code it seems as if no variables are interpolated anywhere during code generation, except for inside the URL.

**Comment by davidsarosap** [comment]
Saw this on https://www.usebruno.com/downloads

![image](https://github.com/user-attachments/assets/2b8c104b-c944-4fb2-b3ec-fdd21732e30a)

It seems that this is not a bug. It is a braking change.

**Comment by davidsarosap** [comment]
It refers to https://github.com/usebruno/bruno/discussions/3072

**Comment by mdm** [comment]
@davidsarosap Maybe that covers your case, where you expect JavaScript to be evaluated. But for the original bug report and my own case it's okay if the variable *content* is treated as a string. It should still be interpolated throughout the request.

**Comment by tpglitch** [comment]
This can be closed as it was fixed with commit https://github.com/usebruno/bruno/commit/f493f72d0e6998802f972a9e31ba71750008e15a

**Comment by stoeps13** [comment]
The mentioned commit is not part of any branch here and this is definitely not working in 3.0.2 it still base64 encodes the variables from user and password when generating the curl call.

**Comment by tpglitch** [comment]
@stoeps13 Maybe a change was made in 3.0.0? I know it says the fix isn't apart of the repo, but it was in some merged PR not too long ago.

**Comment by stoeps13** [comment]
I just found out that the Basic Auth string shows {{user}}:{{pass}} when I reverse the base64 code. So the variable names are still base64 encoded and not the variable content. I understood the change that this should be fixed with the above commit.

**Comment by tpglitch** [comment]
No, this issue is regarding an issue where it literally evaluated to `${something}`
